### ISSUE TITLE
Game State

### ISSUE THREAD

**Issue by sedabull** [sedabull/Greenhorn#11]
The Greenhorn class should maintain some sort of state. As of right now, a single `isRunning()` method can be used to determine whether all of the asynchronous update functions are running, but there should be more that just that. 
* HasNotStarted
* IsRunning
* IsPaused
* IntermediateState (arbitrary number and user defined)
* IsFinished
* UserWon
* UserLost
* etc...

I am unsure of how or when this will be implemented.

**Comment by sedabull** [comment]
I'm thinking the Greenhorn class should have a function called `changeState` that is implemented something like this:
```coffeescript
@changeState = (stateName) ->
    @emit "state-change-from:#{@currentState}"
    @emit "state-change"
    @emit "state-change-to:#{stateName}"
    @currentState = stateName
```

**Comment by sedabull** [comment]
resolved by #21
